client/me/purchases/cancel-purchase/domain-options.jsx

```diff
@@ -116,6 +121,62 @@ const CancelPurchaseDomainOptions = ( {
 		</div>
 	);
 
+	const IncludedDomainTransferPartialRefund = () => (
```

Comment:
This is great that these are their own components, but by creating them _inside_ another component, they are re-created on every render, which creates pretty bad performance. I was surprised to discover that I was the one who had merged this code originally in https://github.com/Automattic/wp-calypso/pull/95065 although it was an OSS PR. I should have noticed this. Anyway, I suppose it's no worse now than it was then, but we should make a follow-up PR to move all these components up one level. I made an issue here:

https://linear.app/a8c/issue/CHE-17/the-cancelpurchasedomainoptions-component-defines-all-its

Reply:
I agree. I thought about it, but I tend to keep the coding style in place.
Also, being defined as `const`, I was expecting the transpiler to be clever enough to prevent rebuilt. Even if I understand they are dependent on the current context of the render.
Nice cath! Thanks for creating the task!